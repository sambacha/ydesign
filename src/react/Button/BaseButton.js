import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { buttonStyle, variant, style } from 'styled-system';
import { StyledBox } from '../Box';
import Icon from '../Icon';

const buttonSize = variant({
  prop: 'size',
  key: 'buttonSizes',
});

const mainColor = style({
  prop: 'mainColor',
  cssProperty: '--main-color',
  key: 'colors',
});

const contrastColor = style({
  prop: 'contrastColor',
  cssProperty: '--contrast-color',
  key: 'colors',
});

const StyledButton = styled(StyledBox)`
  & {
    -webkit-font-smoothing: antialiased;
    appearance: none;
    user-select: none;
    white-space: nowrap;
    text-decoration: none;
    text-align: center;
    overflow: hidden;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    padding: ${props => (props.icononly ? '0' : props.p)};
  }

  &:hover {
    cursor: pointer;
  }

  &:disabled {
    opacity: 0.5;
    pointer-events: none;
  }

  > span.button-text {
    display: inline-flex;
  }

  ${'' /* with icon styles */}
  > div > svg,
  > span.button-text {
    position: relative;
    z-index: 1;
    user-select: none;
    pointer-events: none;
  }

  > div > svg {
    display: block;
  }
  > div:first-child > svg {
    margin-left: -0.5rem;
    margin-right: .5rem;
  }
  > div:last-child > svg {
    margin-left: .5rem;
    margin-right: -0.5rem;
  }
  > div:first-child:last-child > svg {
    margin: 0;
  }

  ${mainColor}
  ${contrastColor}
  ${buttonStyle}
  ${buttonSize}
`;

StyledButton.defaultProps = {
  as: 'button',
};

const ButtonBody = ({ children, icon, iconpos }) => {
  if (icon) {
    return (
      <React.Fragment>
        {icon && !iconpos && <Icon name={icon} />}
        {icon && iconpos == 'left' && <Icon name={icon} />}
        {children && <span className="button-text" children={children} />}
        {icon && iconpos == 'right' && <Icon name={icon} />}
      </React.Fragment>
    );
  } else {
    return <span className="button-text" children={children} />;
  }
};

const Button = React.forwardRef((props, ref) => {
  return <StyledButton {...props} ref={ref} />;
});

Button.defaultProps = {
  // base props
  position: 'relative',
  height: '3rem',
  minWidth: '3rem',
  px: 4,
  py: 0,
  fontSize: 'inherit',
  fontFamily: 'sansSerif',
  fontWeight: 3,
  lineHeight: 1,
  // color props
  color: 'white',
  bg: 'silver',
  mainColor: 'primary',
  contrastColor: 'white',
  border: 'none',
};

Button.propTypes = {
  /**
   * Sets theme
   */
  theme: PropTypes.object,
  /**
   * Sets background color of button
   */
  mainColor: PropTypes.string,
  /**
   * Sets text color of button
   */
  contrastColor: PropTypes.string,
  /**
   * Sets font and spacing size of button
   */
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  /**
   * Sets icon of button
   */
  icon: PropTypes.string,
  /**
   * Sets icon position on button
   */
  iconpos: PropTypes.oneOf(['left', 'right']),
  /**
   * Sets padding on button with icon and no text
   */
  icononly: PropTypes.bool,
};

Button.displayName = 'Button';

export { StyledButton, ButtonBody };

export default Button;
