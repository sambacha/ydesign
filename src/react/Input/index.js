import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { themeGet } from '@styled-system/theme-get'
import { CheckCircle } from '@yfi/icons/es/md'
import { Warning } from '@yfi/icons/es/md'
import FileInput from './FileInput'
import { StyledBox } from '../Box'

const StyledInput = styled(StyledBox).attrs(props => ({
  as: 'input',
}))`
  & {
    appearance: none;
  }

  &:hover {
    box-shadow: 0px 2px 6px rgba(0, 0, 0, 0.3);
  }

  &:focus {
    outline: none;
    border-color: ${themeGet('colors.primary', '#000')};
  }

  &:disabled {
    opacity: 0.4;
    pointer-events: none;
  }

  &[type="color"] {
    min-width: 3rem;
    padding: 0;
    &::-webkit-color-swatch-wrapper {
      display: flex;
      align-items: center;
      justify-content: center;
      padding: 0;
    }
    &::-webkit-color-swatch {
      flex: 0 0 auto;
      height: 2rem;
      width: 2rem;
      border: none;
      border-radius: 100%;
    }
    &::-moz-color-swatch {
      height: 2rem;
      width: 2rem;
      border: none;
      border-radius: 100%;
    }
  }

  .was-validated & {
    &:valid {
      border-color: ${themeGet('colors.success')};
    }
    &:valid ~ svg.icon-valid {
      visibility: visible;
      color: ${themeGet('colors.success')};
    }
    &:invalid {
      border-color: ${themeGet('colors.danger')};
    }
    &:invalid ~ svg.icon-invalid {
      visibility: visible;
      color: ${themeGet('colors.danger')};
    }
  }}
`

const StyledIconWrapper = styled.div`
  & {
    position: relative;
    display: flex;
    align-items: center;
    width: 100%;
  }
  > ${StyledInput} {
    padding-right: 3rem;
    text-overflow: ellipsis;
    white-space: no-wrap;
  }
  > svg {
    position: absolute;
    right: 1rem;
    visibility: hidden;
  }
`

const WithValidationStyle = React.forwardRef((props, ref) => {
  return (
    <StyledIconWrapper>
      <StyledInput {...props} ref={ref} />
      <CheckCircle className={'icon-valid'} />
      <Warning className={'icon-invalid'} />
    </StyledIconWrapper>
  )
})

const Input = React.forwardRef((props, ref) => {
  if (props.type === 'file') {
    return <FileInput {...props} ref={ref} />
  } else {
    return <StyledInput {...props} ref={ref} />
  }
})

const defaultProps = {
  color: 'text',
  bg: 'background',
  fontSize: '1rem',
  lineHeight: 'initial',
  height: '3rem',
  p: 3,
  border: 1,
  borderColor: 'moon-gray',
  borderRadius: 1,
  boxShadow: 1,
}

Input.WithValidationStyle = WithValidationStyle

WithValidationStyle.defaultProps = defaultProps
StyledInput.defaultProps = defaultProps
Input.defaultProps = defaultProps

Input.propTypes = {
  theme: PropTypes.object,
}

Input.displayName = 'Input'

export { StyledInput }

export default Input
