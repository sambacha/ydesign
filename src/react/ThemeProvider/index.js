import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
