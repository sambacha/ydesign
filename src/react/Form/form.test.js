import Form from './';
import Button from '../Button';
import Input from '../Input';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Form component sanity', () => {
  it('has name', () => {
    expect(Form.displayName).toBe('Form');
  });

  it('matches default snapshot', () => {
    const component = render(<Form />);
    expect(component).toMatchSnapshot();
  });

  it('renders children elements', () => {
    const component = render(
      <Form>
        <Form.Field label="Email" width={1}>
          <Form.Input type="email" required width={1} />
        </Form.Field>
        <Form.Field label="Password" width={1}>
          <Form.Input type="password" required width={1} />
        </Form.Field>
        <Form.Check label="Remember me?" mb={3} />
      </Form>
    );
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Form>
          <Form.Field label="Email" width={1}>
            <Form.Input type="email" required width={1} />
          </Form.Field>
          <Form.Field label="Password" width={1}>
            <Form.Input type="password" required width={1} />
          </Form.Field>
          <Form.Check label="Remember me?" mb={3} />
        </Form>
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});

describe('Form validation', () => {
  it('adds was-validated class', () => {
    const component = mount(
      <Form>
        <Form.Field label="Email" width={1}>
          <Form.Input type="email" required width={1} />
        </Form.Field>
      </Form>
    );

    component.setProps({ validated: true });
    expect('validated' in component.props()).toEqual(true);
    expect(
      component
        .find('Box')
        .at(0)
        .hasClass('was-validated')
    ).toEqual(true);
  });
});
