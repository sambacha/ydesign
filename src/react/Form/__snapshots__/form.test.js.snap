// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
}

<form
  class="c0"
/>
`;

exports[`Form component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 16px;
  width: 100%;
}

.c2 {
  box-sizing: border-box;
  font-size: 14px;
  font-weight: 600;
  margin-bottom: 8px;
  line-height: 1.5;
}

.c8 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 16px;
  color: #4E3FCE;
  opacity: 1;
}

.c13 {
  box-sizing: border-box;
  color: #3F3D4B;
  font-size: 14px;
  font-weight: 600;
  line-height: 1;
  margin-left: 4px;
  margin-right: 8px;
}

.c11 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c12 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c6 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c5 {
  box-sizing: border-box;
  width: 100%;
  color: #3F3D4B;
  background-color: #fff;
  font-size: 1rem;
  line-height: initial;
  height: 3rem;
  padding: 16px;
  border: 1px solid transparent;
  border-color: #ccc;
  border-radius: 4px;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c5:hover {
  box-shadow: 0px 2px 6px rgba(0,0,0,0.3);
}

.c5:focus {
  outline: none;
  border-color: #4E3FCE;
}

.c5:disabled {
  opacity: 0.4;
  pointer-events: none;
}

.c5[type="color"] {
  min-width: 3rem;
  padding: 0;
}

.c5[type="color"]::-webkit-color-swatch-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
}

.c5[type="color"]::-webkit-color-swatch {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.c5[type="color"]::-moz-color-swatch {
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.was-validated .c5:valid {
  border-color: #28C081;
}

.was-validated .c5:valid ~ svg.icon-valid {
  visibility: visible;
  color: #28C081;
}

.was-validated .c5:invalid {
  border-color: #DC2C10;
}

.was-validated .c5:invalid ~ svg.icon-invalid {
  visibility: visible;
  color: #DC2C10;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c3 > .c4 {
  padding-right: 3rem;
  text-overflow: ellipsis;
  white-space: no-wrap;
}

.c3 > svg {
  position: absolute;
  right: 1rem;
  visibility: hidden;
}

.c10 {
  position: relative;
  display: inline-block;
}

.c10 > input {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  z-index: 1;
  display: block;
  height: 1.5rem;
  width: 1.5rem;
  margin: 0;
}

.c10 > svg {
  position: absolute;
  z-index: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
  pointer-events: none;
  fill: #CCC;
}

.c10 > svg[name='checked'] {
  display: none;
  z-index: 1;
  fill: #4E3FCE;
}

.c10 > input:checked ~ svg[name='checked'] {
  display: block;
}

.c9 {
  cursor: pointer;
}

.c9:hover input:not(:disabled) ~ svg[name='unchecked'] {
  fill: currentColor;
}

<form
  class="c0"
>
  <label
    class="c1"
    display="inline-flex"
    width="1"
  >
    <div
      class="c2"
      font-size="1"
      font-weight="3"
    >
      Email
    </div>
    <div
      class="c3"
    >
      <input
        class="c4 c5"
        color="text"
        font-size="1rem"
        height="3rem"
        required=""
        type="email"
        width="1"
      />
      <svg
        class="c6 icon-valid"
        color="inherit"
        fill="currentcolor"
        height="24"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <svg
        class="c7 icon-invalid"
        color="inherit"
        fill="currentcolor"
        height="24"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
    </div>
  </label>
  <label
    class="c1"
    display="inline-flex"
    width="1"
  >
    <div
      class="c2"
      font-size="1"
      font-weight="3"
    >
      Password
    </div>
    <div
      class="c3"
    >
      <input
        class="c4 c5"
        color="text"
        font-size="1rem"
        height="3rem"
        required=""
        type="password"
        width="1"
      />
      <svg
        class="c6 icon-valid"
        color="inherit"
        fill="currentcolor"
        height="24"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <svg
        class="c7 icon-invalid"
        color="inherit"
        fill="currentcolor"
        height="24"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
    </div>
  </label>
  <label
    class="c8 c9"
    color="primary"
    display="flex"
    opacity="1"
  >
    <div
      class="c0 c10"
    >
      <input
        color="primary"
        mb="3"
        type="checkbox"
      />
      <svg
        class="c11"
        color="inherit"
        fill="currentcolor"
        height="24"
        name="checked"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 3H5a2 2 0 00-2 2v14a2 2 0 002 2h14a2 2 0 002-2V5a2 2 0 00-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <svg
        class="c12"
        color="inherit"
        fill="currentcolor"
        height="24"
        name="unchecked"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    <div
      class="c13"
      color="text"
      font-size="1"
      font-weight="3"
    >
      Remember me?
    </div>
  </label>
</form>
`;

exports[`Form component sanity renders children elements 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 16px;
  width: 100%;
}

.c2 {
  box-sizing: border-box;
  font-size: 14px;
  font-weight: 600;
  margin-bottom: 8px;
  line-height: 1.5;
}

.c8 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 16px;
  color: #4E3FCE;
  opacity: 1;
}

.c13 {
  box-sizing: border-box;
  color: #3F3D4B;
  font-size: 14px;
  font-weight: 600;
  line-height: 1;
  margin-left: 4px;
  margin-right: 8px;
}

.c11 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c12 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c6 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c5 {
  box-sizing: border-box;
  width: 100%;
  color: #3F3D4B;
  background-color: #fff;
  font-size: 1rem;
  line-height: initial;
  height: 3rem;
  padding: 16px;
  border: 1px solid transparent;
  border-color: #ccc;
  border-radius: 4px;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c5:hover {
  box-shadow: 0px 2px 6px rgba(0,0,0,0.3);
}

.c5:focus {
  outline: none;
  border-color: #4E3FCE;
}

.c5:disabled {
  opacity: 0.4;
  pointer-events: none;
}

.c5[type="color"] {
  min-width: 3rem;
  padding: 0;
}

.c5[type="color"]::-webkit-color-swatch-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
}

.c5[type="color"]::-webkit-color-swatch {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.c5[type="color"]::-moz-color-swatch {
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.was-validated .c5:valid {
  border-color: #28C081;
}

.was-validated .c5:valid ~ svg.icon-valid {
  visibility: visible;
  color: #28C081;
}

.was-validated .c5:invalid {
  border-color: #DC2C10;
}

.was-validated .c5:invalid ~ svg.icon-invalid {
  visibility: visible;
  color: #DC2C10;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c3 > .c4 {
  padding-right: 3rem;
  text-overflow: ellipsis;
  white-space: no-wrap;
}

.c3 > svg {
  position: absolute;
  right: 1rem;
  visibility: hidden;
}

.c10 {
  position: relative;
  display: inline-block;
}

.c10 > input {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  z-index: 1;
  display: block;
  height: 1.5rem;
  width: 1.5rem;
  margin: 0;
}

.c10 > svg {
  position: absolute;
  z-index: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
  pointer-events: none;
  fill: #CCC;
}

.c10 > svg[name='checked'] {
  display: none;
  z-index: 1;
  fill: #4E3FCE;
}

.c10 > input:checked ~ svg[name='checked'] {
  display: block;
}

.c9 {
  cursor: pointer;
}

.c9:hover input:not(:disabled) ~ svg[name='unchecked'] {
  fill: currentColor;
}

<form
  class="c0"
>
  <label
    class="c1"
    display="inline-flex"
    width="1"
  >
    <div
      class="c2"
      font-size="1"
      font-weight="3"
    >
      Email
    </div>
    <div
      class="c3"
    >
      <input
        class="c4 c5"
        color="text"
        font-size="1rem"
        height="3rem"
        required=""
        type="email"
        width="1"
      />
      <svg
        class="c6 icon-valid"
        color="inherit"
        fill="currentcolor"
        height="24"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <svg
        class="c7 icon-invalid"
        color="inherit"
        fill="currentcolor"
        height="24"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
    </div>
  </label>
  <label
    class="c1"
    display="inline-flex"
    width="1"
  >
    <div
      class="c2"
      font-size="1"
      font-weight="3"
    >
      Password
    </div>
    <div
      class="c3"
    >
      <input
        class="c4 c5"
        color="text"
        font-size="1rem"
        height="3rem"
        required=""
        type="password"
        width="1"
      />
      <svg
        class="c6 icon-valid"
        color="inherit"
        fill="currentcolor"
        height="24"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <svg
        class="c7 icon-invalid"
        color="inherit"
        fill="currentcolor"
        height="24"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
    </div>
  </label>
  <label
    class="c8 c9"
    color="primary"
    display="flex"
    opacity="1"
  >
    <div
      class="c0 c10"
    >
      <input
        color="primary"
        mb="3"
        type="checkbox"
      />
      <svg
        class="c11"
        color="inherit"
        fill="currentcolor"
        height="24"
        name="checked"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 3H5a2 2 0 00-2 2v14a2 2 0 002 2h14a2 2 0 002-2V5a2 2 0 00-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <svg
        class="c12"
        color="inherit"
        fill="currentcolor"
        height="24"
        name="unchecked"
        size="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    <div
      class="c13"
      color="text"
      font-size="1"
      font-weight="3"
    >
      Remember me?
    </div>
  </label>
</form>
`;
